addons/mass_mailing/models/mass_mailing.py

```diff
@@ -315,6 +316,20 @@ def _get_source_id(self, cr, uid, context=None):
     def _get_medium_id(self, cr, uid, context=None):
         return self.pool['ir.model.data'].xmlid_to_res_id(cr, uid, 'utm.utm_medium_email')
 
+    # To be removed in saas-8
+    def create(self, cr, uid, vals, context=None):
+        if 'campaign_id' not in vals:
+            vals['campaign_id'] = self.pool['utm.campaign'].create(cr, uid, {'name': vals['name']}, context=context)
+        mass_mailing_campaign_id = super(MassMailingCampaign, self).create(cr, uid, vals, context=context)
+        return mass_mailing_campaign_id
+
+    # To be removed in saas-8
+    def write(self, cr, uid, ids, vals, context=None):
+        for mass_mailing_campaign in self.browse(cr, uid, ids, context=context):
+            if ('name' in vals) and mass_mailing_campaign.campaign_id:
+                self.pool['utm.campaign'].write(cr, uid, [mass_mailing_campaign.campaign_id.id], {'name': vals['name']}, context=context)
```

Comment:
campaign_ids = [mass_mailing_campaign.campaign_id for mass_mailing_campaign in self.browse(ids)]